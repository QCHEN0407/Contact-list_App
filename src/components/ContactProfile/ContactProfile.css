.profile-card {
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    width: 300px;
    position: fixed;
    top: 140px;
    border-radius: 10px;
  }
.profile-card:hover {
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}

.container {
  padding: 2px 16px;
}

.profilePic {
    width: 300px;
    height: 300px;
    object-fit: cover;
    border-top-left-radius: 10px ;
    border-top-right-radius: 10px ;
}

@media screen and (max-width: 1200px) {
  .profile-card {
    position: static;
    margin-bottom: 20px;
  }
  main{
    display: flex;
    flex-direction: column-reverse;
    justify-content: center;
    align-items: center;
  }

}
