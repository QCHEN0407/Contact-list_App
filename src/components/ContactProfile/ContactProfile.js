import React from "react";
import './ContactProfile.css';

function ContactProfile({contact}) {
  return (
    <>
        <div className="card">
            <img data-testid="card-avatar" alt="contactAvatar" className="profilePic" src={contact.avatar} />
            <div className="container">
                <h4 data-testid="card-name"><b>{contact.firstname} {contact.lastname}</b></h4>
                <p data-testid="card-phone">Phone: {contact.phoneNumber}</p>
                <p data-testid="card-email">Email: {contact.email}</p>
            </div>
        </div>
    </>
  );
}

export default ContactProfile;
