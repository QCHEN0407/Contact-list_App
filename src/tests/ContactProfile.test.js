import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import ContactProfile from '../components/ContactProfile/ContactProfile';


const contact = {
  "id": 1,
  "firstname": "Harry",
  "lastname": "Potter",
  "phoneNumber": "5163189902",
  "email":"harryp@gmail.com",
  "avatar":"../img/harry.jpeg"
}

afterEach(cleanup);

it('renders without crashing', ()=> {
  const div = document.createElement("div");
  ReactDOM.render(<ContactProfile contact={contact}/>, div)
})

it('renders name correctly', ()=> {
  const {getByTestId} = render(<ContactProfile contact={contact}/>)
  const element = getByTestId("card-name");
  expect(element).toHaveTextContent("Harry Potter")
})

it('renders phoneNumber correctly', ()=> {
  const {getByTestId} = render(<ContactProfile contact={contact}/>)
  const element = getByTestId("card-phone");
  expect(element).toHaveTextContent("+1 (516) 318-9902")
})

it('renders email correctly', ()=> {
  const {getByTestId} = render(<ContactProfile contact={contact}/>)
  const element = getByTestId("card-email");
  expect(element).toHaveTextContent("harryp@gmail.com")
})

it('renders avatar correctly', ()=>{
  const {getByTestId} = render(<ContactProfile contact={contact}/>)
  const element = getByTestId("card-avatar");
  expect(element).toHaveAttribute("src")
  expect(element).toHaveClass("profilePic")
  expect(element).toHaveAttribute("alt")
})

it('matches snapshot', () => {
  const tree = renderer.create(<ContactProfile contact={contact}/>).toJSON();
  expect(tree).toMatchSnapshot()
})
