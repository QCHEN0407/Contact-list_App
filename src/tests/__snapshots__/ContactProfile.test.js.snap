// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="contactAvatar"
    className="profilePic"
    data-testid="card-avatar"
    src="../img/harry.jpeg"
  />
  <div
    className="container"
  >
    <h4
      data-testid="card-name"
    >
      <b>
        Harry
         
        Potter
      </b>
    </h4>
    <p
      data-testid="card-phone"
    >
      Phone: 
      5163189902
    </p>
    <p
      data-testid="card-email"
    >
      Email: 
      harryp@gmail.com
    </p>
  </div>
</div>
`;
